flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServices.java

```diff
@@ -113,6 +116,11 @@ public ZooKeeperHaServices(
 		this.runningJobsRegistry = new ZooKeeperRunningJobsRegistry(client, configuration);
 
 		this.blobStoreService = checkNotNull(blobStoreService);
+		try {
+			this.submittedJobGraphStore = ZooKeeperUtils.createSubmittedJobGraphs(client, configuration, executor);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
```

Comment:
We should not throw `RuntimeException` but instead a meaningful checked exception.